cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -185,6 +186,24 @@ func MustNew(cfg *config.Config) *Daemon {
 		}, metricsRegistry),
 	}
 
+	fetchCaptiveCoreVersion := func() error {
+		coreClient := daemon.CoreClient()
+		info, err := coreClient.Info(context.Background())
+		if err != nil {
+			return err
+		}
+		config.CaptiveCoreVersion = info.Info.Build
+		return nil
+	}
+
+	backoffStrategy := backoff.NewExponentialBackOff()
+	backoffStrategy.MaxElapsedTime = 4 * time.Second
+	err = backoff.Retry(fetchCaptiveCoreVersion, backoffStrategy)
+	if err != nil {
```

Comment:
Does this mean the backoff strategy itself failed? imo yes I think it's safe to stop RPC here since it means something is very wrong with Captive Core, but maybe extend the `MaxElapsedTime` to 10s or even 30s.

Reply:
captive core is not running at this point in the code which is why the integration tests are probably failing. captive-core only starts running during the ingestion service

Reply:
As per this [discussion](https://github.com/stellar/soroban-rpc/pull/132#discussion_r1570670897) setting core version in daemon won't work, so removed this logic. Please note I am going back to initial approach of calling /info endpoint. Adding the support of caching along with new [function](https://github.com/stellar/soroban-rpc/pull/132#discussion_r1570930941) in core impl will be done in follow up PR. But for now I want to push this to prod to have something for clients to rely on :)